### PR TITLE
[10] As a user, I can search a imported keyword

### LINKED ISSUE #10: [Frontend] As a user, I can search a imported keyword
Allow users to query all the stored data across all their uploaded keywords.

Examples of the SQL queries which users can run:

- How many URLs contain the word “technology” in AdWords.
- How many times a specific URL shows up in stored search results.
- How many keywords have URLs in stored search results with 2 or more “/” or 1 or more “>”.

## Acceptance criteria
- [ ] Display an input for the user to enter the keyword.
- [ ] Display search button 
- [ ] When the user enters directly on the input, it should be able to search the same behavior with a click button.

### PR DESCRIPTION
**Pull request by byhbt**
https://github.com/byhbt/kwtool/issues/10

## What happened

- Add the search box on top of the listing keywords table.
- Implement basic search by the **keyword** phrase only. We will implement the search of crawler results in [another ticket](https://github.com/byhbt/kwtool/projects/1#card-60618902).

Notice:
There is some backend code in this PR because it's small to separate into a dedicated backend PR, that's why I keep it here, to make the search feature work.

## Insight

n/a
 
## Proof Of Work

search-feature-demo

### CODE REVIEW

File: lib/kwtool_web/templates/keyword/index.html.eex [hunk 1]
```diff
@@ -16,8 +18,8 @@
           <td><%= keyword.phrase %></td>
           <td><%= keyword.inserted_at %></td>
           <td><span class="status-badge--added"><%= keyword.status %></span></td>
-          <td>
-            <span><%= link "View", class: "button-detail", to: Routes.keyword_path(@conn, :show, keyword) %></span>
+          <td class="keyword-list__action-column">
+            <span><%= link "View", class: "btn btn-sm btn-secondary", to: Routes.keyword_path(@conn, :show, keyword) %></span>
```
Comment (byhbt) [inline lib/kwtool_web/templates/keyword/index.html.eex:22]: @hanam1ni agreed with your comment on this button, I updated it to use the bootstrap class instead.

File: test/kwtool/crawlers_test.exs [hunk 2]
```diff
@@ -67,6 +67,30 @@ defmodule Kwtool.CrawlersTest do
...
+      insert(:keyword, custom_keyword_attrs)
+
+      search_query = %{"query" => "test search param"}
+      {keywords, _pagination} = Crawlers.paginate_user_keywords(created_user, search_query)
+
+      assert length(keywords) == 1
+      assert Enum.at(keywords, 0).phrase == custom_keyword_attrs.phrase
+    end
+
+    test "returns a empty list when given a not matched query params" do
```
Comment (hanam1ni) [inline test/kwtool/crawlers_test.exs]:
```suggestion
    test "returns an empty list when query params not match any keyword" do
```

**Review by hanam1ni — APPROVED** [review]
One small comment

File: test/kwtool/crawlers_test.exs (hunk 2 above)
Comment (byhbt) [reply test/kwtool/crawlers_test.exs]: Updated! Thanks 👍

File: lib/kwtool_web/templates/keyword/index.html.eex [hunk 3]
```diff
@@ -1,7 +1,9 @@
-<h1>Listing Keywords</h1>
+<h1 class="page-title">Listing Keywords</h1>
 
 <%= if Enum.any?(@keywords) do %>
-  <table class="table">
+  <%= render "_search-form.html", Map.put(assigns, :action, Routes.keyword_path(@conn, :index)) %>
```
Comment (rosle) [inline lib/kwtool_web/templates/keyword/index.html.eex]: I think the file name should be `_search_form` (with underscore)

File: test/kwtool/crawlers_test.exs [hunk 4]
```diff
@@ -67,6 +67,30 @@ defmodule Kwtool.CrawlersTest do
                per_page: 15
              } == pagination
     end
+
+    test "returns a list of matched keywords when given a query params" do
```
Comment (rosle) [inline test/kwtool/crawlers_test.exs]: Is it supposed to be 🤔 

1)
```suggestion
    test "returns a list of matched keywords when given a matched query params" do
```

or

2)
```suggestion
    test "returns a list of matched keywords when given a query param" do
```
If it's 2, probably remove -s on line #85 as well

File: test/kwtool_web/features/keyword_page/view_keyword_page_test.exs [hunk 5]
```diff
@@ -27,7 +58,9 @@ defmodule KwtoolWeb.HomePage.ViewKeywordPageTest do
     |> assert_has(Query.text("Result for"))
   end
 
-  feature "redirects to listing keywords page when given other user keyword", %{session: session} do
+  feature "redirects to the listing keywords page when given a keyword of other user", %{
```
Comment (rosle) [inline test/kwtool_web/features/keyword_page/view_keyword_page_test.exs]:
```suggestion
  feature "redirects to the listing keywords page when given a keyword of another user", %{
```

Comment (byhbt) [reply test/kwtool_web/features/keyword_page/view_keyword_page_test.exs]: Updated! Thanks 👍

File: test/kwtool/crawlers_test.exs (hunk 4 above)
Comment (byhbt) [reply test/kwtool/crawlers_test.exs]: i updated follow the (2) approach! Thanks 👍,

**Review by rosle — APPROVED** [review]
👍

**Pull request merged**